rest_framework/filters.py

```diff
@@ -104,8 +105,13 @@ def filter_queryset(self, request, queryset, view):
         for search_term in self.get_search_terms(request):
             or_queries = [models.Q(**{orm_lookup: search_term})
                           for orm_lookup in orm_lookups]
-            queryset = queryset.filter(reduce(operator.or_, or_queries)).distinct()
+            queryset = queryset.filter(reduce(operator.or_, or_queries))
 
+        if settings.DATABASES[queryset.db]["ENGINE"] != "django.db.backends.oracle":
+            queryset = queryset.distinct()
+        else:
+            pk_list = queryset.values_list('pk', flat=True)
+            queryset = view.model.objects.filter(pk__in=set(pk_list))
```

Comment:
Using `view.model.objects` here feels a little bit brittle. I think we can avoid needing that. Does something like the following make sense?...

    if settings.DATABASES[queryset.db]["ENGINE"] != "django.db.backends.oracle":
        return queryset.filter(reduce(operator.or_, or_queries)).distinct()
    else:
        pk_list = queryset.filter(reduce(operator.or_, or_queries)).values_list('pk', flat=True)
        return queryset.filter(pk_in=set(pk_list))